Use foundry output files in for tests and remove old docker build

ClementWalter opened the pull request:
## Pull request type 

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Solidity files are duplicated and foundry is not used to compile files used for testing.

## What is the new behavior?

Use foundry for building all the solidity files in the projet and make these builds available for integration tests.

## Other information

I removed the scripts using docker that became useless since it's a single command with forge, so added it directly to the Makefile

Time spent on this PR:  0.5

Eikix commented on tests/integration/solidity_contracts/Solmate/test_erc721.py:
```diff
@@ -14,7 +12,7 @@ async def test_should_set_name_and_symbol(
         ):
             erc_721 = await deploy_solidity_contract(
-                "ERC721", "ERC721", "Kakarot NFT", "KKNFT", caller_address=1
+                "Solmate", "ERC721", "Kakarot NFT", "KKNFT", caller_address=1
```
why does the test python file live inside the solidity_contracts folder?

ClementWalter replied:
cc @matthieuauger ; it was decided to ease the testing of .sol files, keeping them close to the test file

ClementWalter commented:
> why are we deleting PlainOpcodes.sol and Counter.sol?

they had been duplicated in forge pr. I just removed the dups ones and moved all the existing sol files in the same `tests/integrations/solidity_contracts` folder where it's easier (atm) to have them for writing tests

Eikix reviewed (approved):
lgtm
